File: Easy/108. Convert Sorted Array to Binary Search Tree/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode sortedArrayToBST(int[] arr) {

      return   createBST(arr, 0,arr.length-1);
    }
    public static TreeNode createBST(int arr[], int si, int end){
          if(si> end){
            return null;
        }
        int mid=(si+end)/2;
        TreeNode root= new TreeNode(arr[mid]);
        root.left= createBST(arr, si, mid-1);
        root.right= createBST(arr, mid+1, end);
        return root;

    }
}

File: Easy/108. Convert Sorted Array to Binary Search Tree/Solution.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def sortedArrayToBST(self, nums):
        if not nums:
            return None
        head = self.helper(nums, 0, len(nums) - 1)
        return head

    def helper(self, nums, low, high):
        if low > high:
            return None
        mid = low + (high - low) // 2
        node = TreeNode(nums[mid])
        node.left = self.helper(nums, low, mid - 1)
        node.right = self.helper(nums, mid + 1, high)
        return node

